### PR TITLE
mkctr: add "local" target

### PR DESCRIPTION
**Pull request by maisem**
This pushes to the locally running Docker instance.

### CODE REVIEW

File: mkctr.go [hunk 1]
```diff
@@ -158,10 +160,29 @@ func fetchBaseImage(baseImage string, opts ...remote.Option) (*remote.Descriptor
...
+	}
+	return false
+}
+
 func verifyPlatform(p v1.Platform, target string) error {
 	if p.OS != "linux" {
 		return fmt.Errorf("unsupported OS: %v", p.OS)
 	}
+	if target == "local" && canRunLocal(p) {
+		return fmt.Errorf("not required for target %q", target)
```
Comment (willnorris) [inline mkctr.go:184]: I'm not quite clear on what this error is telling me and what I would do about it if I saw it.  I've specified `--target=local`, but it's not required for the architecture I'm on?  Why would that be an error state?  (I'm sure I'm misunderstanding)

**Review by willnorris — APPROVED** [review]
One comment about a confusing (to me) error message.  Otherwise, LGTM

**Comment by irbekrm** [comment]
@maisem with what base where you able to test this? I didn't spend much time on it, but it seemed that multi-arch bases fail because mkctr then attempts to create a multi-arch image which fails the local target, and single arch bases generally fail on mkctr [here](https://github.com/tailscale/mkctr/blob/main/mkctr.go#L197) because they don't have platform description (at least the ones I tried)

**Comment by maisem** [comment]
I tested it using a modified version of the build_docker.sh from the tailscale repo

**Review by irbekrm — APPROVED** [review]

**Pull request merged**